test/e2e/vulnerability/vulnerability_test.go

```diff
@@ -0,0 +1,116 @@
+	googleApplicationKeyEnvName     = "GOOGLE_APPLICATION_CREDENTIALS"
+	azureAuthKeyTestEnvName         = "AZURE_AUTH_TEST_SECRET"
+	azureAuthKeyEnvName             = "AZURE_AUTH_LOCATION"
+)
+
+var _ = Describe("Vulnerabilty", func() {
+
+	BeforeSuite(func() {
+		terrascanBinaryPath = helper.GetTerrascanBinaryPath()
+	})
+
+	BeforeEach(func() {
+		outWriter = gbytes.NewBuffer()
+		errWriter = gbytes.NewBuffer()
+	})
+
+	AfterEach(func() {
+		outWriter = nil
+		errWriter = nil
+	})
+
+	Describe("resource with conatiner images to be scanned for vulnerabilities", func() {
+		resourceVulnerabilityIacRelPath := filepath.Join(iacRootRelPath, "resource_for_vulnerability_scan")
+		Context("resource with aws ecr registry image used which has vulnerabilities", func() {
+			iacDir := filepath.Join(resourceVulnerabilityIacRelPath, "aws_ecr_registry_used_in_resource")
+			It("should display vulnerabilities and exit with status code 0 since no violations found", func() {
+				scanArgs := []string{"-p", policyRootRelPath, "-i", "terraform", "-d", iacDir, "-o", "json"}
+				vulnUtils.RunScanAndVerifyVulnerabilityOutputCount(terrascanBinaryPath, helper.ExitCodeZero, false, true, outWriter, errWriter, scanArgs...)
+			})
+		})
+		Context("resource with azure registry image used which has vulnerabilities", func() {
+			iacDir := filepath.Join(resourceVulnerabilityIacRelPath, "azure_registry_used_in_resource")
+			JustBeforeEach(func() {
+				data := os.Getenv(azureAuthKeyTestEnvName)
+				if data != "" {
+					tempFile, err = ioutil.TempFile("", "azure.auth")
+					Expect(err).NotTo(HaveOccurred())
+					tempFile.Write([]byte(data))
+					os.Setenv(azureAuthKeyEnvName, tempFile.Name())
+				}
+			})
+			JustAfterEach(func() {
+				if tempFile != nil && tempFile.Name() != "" {
+					os.Remove(tempFile.Name())
+				}
+			})
+			It("should display vulnerabilities and exit with status code 0 since no violations found", func() {
+				scanArgs := []string{"-p", policyRootRelPath, "-i", "terraform", "-d", iacDir, "-o", "json"}
+				vulnUtils.RunScanAndVerifyVulnerabilityOutputCount(terrascanBinaryPath, helper.ExitCodeZero, false, true, outWriter, errWriter, scanArgs...)
+			})
+		})
+		Context("resource with google registry image used which has vulnerabilities", func() {
+			iacDir := filepath.Join(resourceVulnerabilityIacRelPath, "google_gcr_registry_used_in_resource")
+			JustBeforeEach(func() {
+				data := os.Getenv(googleApplicationKeyTestEnvName)
+				if data != "" {
+					tempFile, err = ioutil.TempFile("", "app-key.json")
+					Expect(err).NotTo(HaveOccurred())
+					tempFile.Write([]byte(data))
+					os.Setenv(googleApplicationKeyEnvName, tempFile.Name())
+				}
+			})
+			JustAfterEach(func() {
+				if tempFile != nil && tempFile.Name() != "" {
+					os.Remove(tempFile.Name())
+				}
+			})
+			It("should display vulnerabilities and exit with status code 0 since no violations found", func() {
```

Comment:
code nits: we probably should name the spec as `should display vulnerabilities and scan summary` and add comment related to the exit code before the `RunScanAndVerifyVulnerabilityOutputCount` function is called.